src/app/activity/bounty/Publish.ts

```diff
@@ -43,21 +45,26 @@ export const publishBounty = async (publishRequest: PublishRequest): Promise<any
 		});
 		await previewMessage.delete();
 	}
+
+	await updateMessageStore(dbBountyResult, bountyMessage);
 	
 	return;
 }
 
-const getDbHandler = async (bountyId: string, guildId: string): Promise<[BountyCollection, CustomerCollection]> => {
+const getDbHandler = async (bountyId: string, guildId: string, request: PublishRequest): Promise<[BountyCollection, CustomerCollection]> => {
     Log.debug(`Entered get DbHandler for publish`);
     const db: Db = await MongoDbUtils.connect('bountyboard');
 	const dbCollectionBounties = db.collection('bounties');
     const dbCollectionCustomers = db.collection('customers');
+
+	const status = request.clientSyncRequest ? 'Open' : 'Draft';
+
     const dbCustomerResult: CustomerCollection = await dbCollectionCustomers.findOne({
 		customerId: guildId
 	});
     const dbBountyResult: BountyCollection = await dbCollectionBounties.findOne({
 		_id: new mongo.ObjectId(bountyId),
-		status: 'Draft',
+		status: status,
```

Comment:
You're 100% right, it isn't. We explicitly screen for the status in the validation handler. The reason i've been comfortable with keeping `status` as part of queries is because of a really sensible comment you made a while back about race conditions between two clients hitting the same db. 

It's an extra line, but if there's an update to the data model between the validation handler and the activity, this will catch that and we'll error out of the activity before any writes are performed.

Reply:
Makes sense